service/terraform-plugin/src/test/java/com/solace/maas/ep/event/management/agent/plugin/terraform/TerraformCommandIT.java

```diff
@@ -0,0 +1,357 @@
+
+        // Setup the output
+        setupLogMock(deleteQueueTfLogs);
+
+        terraformManager.execute(terraformRequest, command, Map.of());
+
+        // Validate that the plan api is called
+        verify(terraformClient, times(1)).plan(any());
+        verify(terraformClient, times(1)).apply(any());
+
+        // Check the responses
+        for (CommandBundle commandBundle : terraformRequest.getCommandBundles()) {
+            for (Command tfCommand : commandBundle.getCommands()) {
+
+                CommandResult result = tfCommand.getResult();
+                assertEquals(JobStatus.success, result.getStatus());
+                assertTrue(result.getLogs().get(2).get("message").toString().contains("Destruction complete after"));
+                assertTrue(result.getLogs().get(3).get("message").toString().contains("Destruction complete after"));
+            }
+        }
+    }
+
+    private static Command generateCommand(String tfCommand, String body) {
+        return generateCommand(tfCommand, body, false);
+    }
+
+    private static Command generateCommand(String tfCommand, String body, Boolean ignoreResult) {
+        return generateCommand(tfCommand, body, ignoreResult,
+                Map.of("Content-Type", "application/hcl",
+                        "Content-Encoding", "base64"));
+    }
+
+    private static Command generateCommand(String tfCommand, String body, Boolean ignoreResult, Map<String, String> parameters) {
+        return Command.builder()
+                .body(Optional.ofNullable(body)
+                        .map(b -> Base64.getEncoder().encodeToString(b.getBytes(UTF_8)))
+                        .orElse(""))
+                .command(tfCommand)
+                .commandType(CommandType.terraform)
+                .ignoreResult(ignoreResult)
+                .parameters(parameters)
+                .build();
+    }
+
+
+    private static CommandRequest generateCommandRequest(Command commandRequest) {
+        return CommandRequest.builder()
+                .commandBundles(List.of(
+                        CommandBundle.builder()
+                                .executionType(ExecutionType.serial)
+                                .exitOnFailure(false)
+                                .commands(List.of(commandRequest))
+                                .build()))
+                .correlationId("234")
+                .context("app123")
+                .serviceId("ms1234")
+                .build();
+    }
+
+    private void setupLogMock(List<String> logs) {
+        doAnswer(invocation -> {
+            Object arg0 = invocation.getArgument(0);
+            Consumer<String> listener = (Consumer<String>) arg0;
+            logs.forEach(listener);
+            return null;
+        }).when(terraformClient).setOutputListener(any());
+    }
+
+
+    public static String getResourceAsString(Resource resource) {
```

Comment:
Does this method and the one below need to be `public static` or would `private` suffice?

Reply:
Private will do. Updating...